keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -715,30 +716,24 @@ to describe a selector based policy that applies to a multi-cluster service.
 
 _Optional, but recommended._
 
-MCS aims to align with the existing [service DNS
+The full specification for Multicluster Service DNS is in this KEP's [specification.md](specification.md). MCS aims to align with the existing [service DNS
 spec](https://github.com/kubernetes/dns/blob/master/docs/specification.md). This
-section assumes familiarity with in-cluster Service DNS behavior.
-```
-<<[UNRESOLVED]>>
-Full DNS spec needed prior to Beta graduation following:
-https://github.com/kubernetes/dns/blob/master/docs/specification.md
-<<[UNRESOLVED]>>
-```
-When a `ServiceExport` is created, this will cause a domain name for the
+section provides an overview of the multicluster DNS specification and its rationale, and assumes familiarity with in-cluster Service DNS behavior.
+
+In short, when a `ServiceExport` is created, this will cause a domain name for the
 multi-cluster service to become accessible from within the clusterset. The
-domain name will be `<service>.<ns>.svc.clusterset.local`. 
+domain name will be `<service>.<ns>.svc.clusterset.local`. This domain name operates differently depending on whether the `ServiceExport` refers to a ClusterSetIP or Headless service:
 
-**ClusterSetIP services:** Requests to this domain name from within an importing
+  * **ClusterSetIP services:** Requests to this domain name from within an importing
 cluster will resolve to the clusterset IP, which points to endpoints for pods
 within the underlying `Service`(s) across the clusterset.
-
-**Headless services:** Within an importing cluster, the clusterset domain name
+  * **Headless services:** Within an importing cluster, the clusterset domain name
 will have multiple `A`/`AAAA` records, each containing the address of a ready
 endpoint of the headless service. `<service>.<ns>.svc.clusterset.local` will
 resolve to the set of all ready pod IPs for the service.
 
-Pods backing a clusterset service may be addressed individually using the
-`<hostname>.<clustername>.<svc>.<ns>.svc.clusterset.local` format. Necessary
+In addition, other resource records are included to conform to in-cluster Service DNS behavior. SRV records are included to support known use cases such as VOIP, Active Directory, and etcd cluster bootstrapping. PTR records are required at the multicluster DNS level only for implementations that do not already have PTR records for clusterset IPs from the in-cluster DNS specification (for example, implementations that utilize a "dummy" in-cluster service to represent the clusterset IP). Pods backing a Headless service may be addressed individually using the
+`<hostname>.<clusterid>.<svc>.<ns>.svc.clusterset.local` format; necessary
```

Comment:
```suggestion
`<podname>.<clusterid>.<svc>.<ns>.svc.clusterset.local` format; necessary
```
?

Reply:
So I understand this would cover a use case like this: 
https://github.com/raffaelespazzoli/openshift-modern-availability/blob/master/cockroachdb/values.templ.yaml#L24

Where every single pod needs to be addressed to form a cluster (in the application level), but then the final user of the application can still access <svc>.<ns>.svc.clusterset.local

Reply:
Hi! Yes, exactly on that use case (and more broadly should be the same as [StatefulSets and Headless Services in submariner.io docs](https://submariner.io/operations/usage/#test-statefulset-and-headless-service)).

Re: `<podname>`, I'm keeping `<hostname>` as 
a) it is the same terminology as defined and then used in the [single-cluster DNS spec](https://github.com/kubernetes/dns/blob/master/docs/specification.md#21---definitions) and 
b) it should not always be the pod name (though it likely often is, since that is the [default hostname](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields)), as implementations should prefer any directly specified hostname value.

However I think this is a really common colloquialism (I painfully unlearned `podname=hostname` for myself in the course of this project, haha) so I will reproduce the definition of `hostname` in this spec directly so it's easy to find.

Reply:
Added hostname definition to this file directly in f609fdc